src/main/java/org/mule/extension/file/common/api/command/WriteCommand.java

```diff
@@ -53,4 +53,20 @@ void write(String filePath, InputStream content, FileWriteMode mode, boolean loc
   default void write(String filePath, InputStream content, FileWriteMode mode, boolean lock, boolean createParentDirectory) {
     write(filePath, content, mode, lock, createParentDirectory, null);
   }
+
+  /**
+   * Writes a file under the considerations of {@link FileSystem#write(String, InputStream, FileWriteMode, boolean, boolean)}
+   *
+   * @param filePath the path of the file to be written
+   * @param content the content to be written into the file
+   * @param mode a {@link FileWriteMode}
+   * @param lock whether or not to lock the file
+   * @param createParentDirectory whether or not to attempt creating the parent directory if it doesn't exist.
+   * @param lockTimeout time in nanoseconds that the operation will spend trying to lock the file.
+   * @throws IllegalArgumentException if an illegal combination of arguments is supplied
+   */
+  default void write(String filePath, InputStream content, FileWriteMode mode, boolean lock, boolean createParentDirectory,
+                     long lockTimeout) {
+    write(filePath, content, mode, lock, createParentDirectory, null);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "write"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-file-commons&pullRequest=56&issues=AWm2SxsiotIXnmsUPRcE&open=AWm2SxsiotIXnmsUPRcE)

[ISSUE_KEY]: AWm2SxsiotIXnmsUPRcE